### DISCUSSION TITLE
Unable to install Node version v16.14.0

### DISCUSSION THREAD

**Discussion by anniewey** [codemagic-ci-cd/codemagic-docs#2658] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Failed at `Preparing Build Machine` today. Last build was run successfully 6 days ago. No changes been made to the yaml file

```

== Use Node version v16.14.0 ==
> n v16.14.0


Build failed :|
Unable to install Node version v16.14.0
```

### Expected results

Expected to be able to install node

### Actual results

Failing at install node step

### Build id (optional)

66ab3689419b16142e4c711c

**Comment by dtrdic** [comment, marked as answer]
hey @anniewey there could to be an issue on node side that caused intermittent problems with file serving through Cloudflare. To avoid this kind of issues in the future, if your project is compatible, we recommend using the pre-installed version on build machine.
​
​node 20.12.2
npm 10.5.0
​
​https://docs.codemagic.io/specs/versions-macos/#:~:text=node%2020.12.2,npm%2010.5.0
​
